### PR TITLE
(GH-14) Add backup cleanup function

### PR DESCRIPTION
**Pull request by stefangweichinger**
Initial draft, to be tested. Relates to #14 .

### CODE REVIEW

File: backup.sh [hunk 1]
```diff
@@ -34,4 +34,7 @@ do
   fi
 done
 
-echo "Backup process finished at $(date "+%Y-%m-%d %H:%M:%S")"
\ No newline at end of file
+find $BACKUPDIR -type f -name '*.bak' -mtime +"$AGE_BACKUPS" -exec rm {} \;
```
Comment (stefangweichinger) [inline backup.sh:37]: I am extra cautious here and avoid something like finding '*'. You never know what else writes to that directory in the future, better to be explicit.

File: README.md [hunk 2]
```diff
@@ -42,6 +42,7 @@ These environment variables are supported:
 | DB_NAMES             |               | Names of the databases for which a backup should be created.                                                                                                                                                                     |
 | TZ                   |               | Timezone to use.                                                                                                                                                                                                                 |
 | CRON_SCHEDULE        | `0 1 * * sun` | Cron schedule for running backups. NOTE: There is no check if there's already a backup running when starting the backup job. Therefore time interval needs to be longer than the maximum expected backup time for all databases. |
+| AGE_BACKUPS          | `7` | Number of days to keep backups in backup directory |
```
Comment (eoehen) [inline README.md:46]: Documentation for BACKUPDIR (or BACKUP_DIR) is missing.

File: backup.sh (hunk 1 above)
Comment (eoehen) [inline backup.sh:37]: You should use the same naming scheme as for creating.

Look at...
`BAK_FILENAME=/backup/$CURRENT_DATE.$CURRENT_DB.bak`
[](
https://github.com/bbtsoftware/docker-mssql-backup/blob/f5aef0761c766f5cd64a1a9510be879fdbb4ebfa)/backup.sh#L14

I would use 
`find $BACKUPDIR -type f -name '*.$CURRENT_DB.bak' -mtime +"$AGE_BACKUPS" -exec rm {} \;`

... for each database (CURRENT_DB).

**Review by eoehen — CHANGES_REQUESTED** [review]

**Comment by stefangweichinger** [comment]
That find/rm doesn't work that way. The script runs in the backup-container, but the $BACKUPDIR only exists in the db-container.
We would have to mount the target volume in both containers to make that work.

**Review by eoehen — CHANGES_REQUESTED** [review]
Many thanks for your thoughts. You are right.

**Comment by eoehen** [comment]
> 
> 
> That find/rm doesn't work that way. The script runs in the backup-container, but the $BACKUPDIR only exists in the db-container.
> We would have to mount the target volume in both containers to make that work.

File: README.md [hunk 3]
```diff
@@ -42,6 +42,9 @@ These environment variables are supported:
 | DB_NAMES             |               | Names of the databases for which a backup should be created.                                                                                                                                                                     |
 | TZ                   |               | Timezone to use.                                                                                                                                                                                                                 |
 | CRON_SCHEDULE        | `0 1 * * sun` | Cron schedule for running backups. NOTE: There is no check if there's already a backup running when starting the backup job. Therefore time interval needs to be longer than the maximum expected backup time for all databases. |
+| BACKUP_CLEANUP      | `false` | Set to "true" if you want to let the cronjob remove files older than $BACKUP_AGE days |
+| BACKUP_AGE          | `7` | Number of days to keep backups in backup directory |
+| BACKUP_DIR          | `/backup` | Backup directory in the MSSQL container  |
```
Comment (eoehen) [inline README.md:47]: Why do we need the possibility to specify the backup directory BACKUP_DIR?

**Review by eoehen — CHANGES_REQUESTED** [review]

File: README.md (hunk 3 above)
Comment (stefangweichinger) [reply README.md:47]: Flexibility?
By making it a variable you can set a specific path without rebuilding your image.
As you like, it's just a suggestion.

Comment (eoehen) [reply README.md:47]: I would remove them. I see no use for it at the moment. For the volume mapping it is easier we have `/backup` fix defined folders inside the containers.

**Review by eoehen — CHANGES_REQUESTED** [review]

File: README.md (hunk 3 above)
Comment (stefangweichinger) [reply README.md:47]: So you would remove that variable and want me to edit the PR? ;-)

Comment (eoehen) [reply README.md]: Sorry I was unclear. It would be nice if you could do that in your PR. I will test your solution as soon as possible.

Comment (stefangweichinger) [reply README.md]: done already a few hours ago in : https://github.com/bbtsoftware/docker-mssql-backup/pull/17/commits/cca6df533340d2e77873a580bc5717a71950d5b6

I am on vacation from tomorrow.

Comment (eoehen) [reply README.md]: Thank you very much for your contribution. 👍 
I will finish and test your pr. Enjoy your vacation in switzerland. :-)

**Comment by eoehen** [comment]
@stefangweichinger your changes have been merged, thanks for your contribution 👍

**Pull request merged**